src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+}
+
+hal::celsius bmp180::temperature()
+{
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // return just the temperature data
+  return temperature_data.temperature;
+}
+
+bmp180::pressure_results bmp180::pressure(int sample_amount)
+{
+  // clamp sample amount to determined maximum
+  sample_amount = std::clamp(sample_amount, 1, m_maximum_samples);
+
+  computation_variables variables;
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // tell sensor to start conversion for pressure
+  hal::byte control_setting =
+    (static_cast<hal::byte>(m_oversampling_setting) << 6) | 0x34;
+  std::array<hal::byte, 2> sensor_config_buffer = {
+    control_measurement_register()[0], control_setting
+  };
+
+  hal::byte oversampling_setting =
+    static_cast<hal::byte>(m_oversampling_setting);
+  std::int32_t summed_uncompensated_pressures = 0,
+               average_uncompensated_pressure = 0;
+
+  // loop to wait for conversion to complete
+  int conversion_finished;
+  for (int i = 0; i < sample_amount; i++) {
+    conversion_finished = 0;
+    hal::write(*m_i2c, m_address, sensor_config_buffer, hal::never_timeout());
+    while (conversion_finished == 0) {
+      conversion_finished = conversion_status(m_i2c, m_address);
+    }
+
+    // grab the raw data from the conversion and store it
+    auto uncompensated_data_buffer =
+      hal::write_then_read<3>(*m_i2c,
+                              m_address,
+                              uncompensated_output_data_register(),
+                              hal::never_timeout());
+    std::int32_t uncompensated_pressure =
+      ((uncompensated_data_buffer[0] << 16) |
+       (uncompensated_data_buffer[1] << 8) | uncompensated_data_buffer[2]) >>
+      (8 - oversampling_setting);
+    // accumulate the raw samples together to later find average
+    summed_uncompensated_pressures += uncompensated_pressure;
+  }
+  average_uncompensated_pressure =
+    summed_uncompensated_pressures / sample_amount;
+
+  // convert the raw pressure data into Pascals in steps of 1Pa
```

Comment:
Add a comment explain why you did not use bit_modify here and where in the datasheet these equations are located.

Reply:
And that this is written exactly as it was spelled out in the datasheet.